sockeye/data_io.py

```diff
@@ -437,12 +437,14 @@ def load(self,
         num_tokens_target = 0
         num_pad_source = 0
         num_pad_target = 0
+        map_buckets2sentence_ids = defaultdict(lambda: defaultdict(int))
 
         # Bucket sentences as padded np arrays
-        for sources, target in zip(zip(*sources_sentences), target_sentences):
+        for sentence_id, (sources, target) in enumerate(zip(zip(*sources_sentences), target_sentences)):
             source_len = len(sources[0])
             target_len = len(target)
             buck_index, buck = get_parallel_bucket(self.buckets, source_len, target_len)
+            ## TODO does this ever happen if we use maxlens from test set? check
```

Comment:
I was wondering what should happen in case we have maxlen given on CLI and source or target are > args.maxlen.. in this case, get_parallel_bucket will return None, should we drop the sentence pair..? What should be the output, empty line, None..? If maxlen is not given as argument, the data_iterator gets the maximum lengths from the data set itself, so get_parallel_bucket will always return a bucket index.. could also remove maxlen from arguments for scoring?

Reply:
that's a good question. For translation we split a sentence into multiple smaller pieces if it exceeds the maximum input length and translate them separately. For scoring it would make sense to either do something similar or reject those sentences, return an empty line and document this behavior in the CLI.